server/datastore/mysql/maintained_apps.go

```diff
@@ -128,6 +130,11 @@ WHERE NOT EXISTS (
 	)
 )`
 
+	// build the count statement before adding the pagination constraints to the
+	// default stmt.
+	dbReader := ds.reader(ctx)
+	getAppsCountStmt := fmt.Sprintf(`SELECT COUNT(DISTINCT s.id) FROM (%s) AS s`, stmt)
```

Comment:
Looks like it's the total count, so I think it could be returned in the `TotalCount` field instead of a new returned value (not critical, so up to you in case that PR is urgent to merge). That field does not get rendered in JSON, it's just to "carry" the count up to the endpoint handler where you still have to assign it to the response struct explicitly.

Reply:
For a similar approach in another endpoint, "ListHostSoftware" is implemented like that and the total count gets assigned to the response payload here: https://github.com/fleetdm/fleet/blob/e16076851464c4abd015271f9c2e32ded82c2328/server/service/hosts.go#L2587-L2594